### PR TITLE
Modification to make owner of region inside of it if the box contains it

### PR DESCRIPTION
**Pull request by rafaelgmota**
If box of region contains the owner he will be included at region, with this we decrease the number of generated regions

### CODE REVIEW

File: src/rbr/RBR.java [hunk 1]
```diff
@@ -129,10 +129,25 @@ public void regionsComputation() {
         assert reachabilityIsOk();
     }
 
+    private boolean isOwnerInside(Region r, Vertex owner) {
+        Range box = r.box();
```
Comment (rmilfont) [inline src/rbr/RBR.java]: I think that you can remove this line and put in the return statement

File: src/rbr/RBR.java [hunk 2]
```diff
@@ -129,10 +129,25 @@ public void regionsComputation() {
...
+        return box.contains(Integer.valueOf(owner.name().split("\\.")[0]), Integer.valueOf(owner.name().split("\\.")[1]));
+    }
+
+    Region adjustRegionWithOwner(Region r, Vertex owner) {
+        if(isOwnerInside(r, owner)) {
+            Set<Vertex> destinations = new HashSet<>();
```
Comment (rmilfont) [inline src/rbr/RBR.java]: You don´t have to do this new.

File: src/rbr/RBR.java [hunk 3]
```diff
@@ -129,10 +129,25 @@ public void regionsComputation() {
...
+            return new Region(r.inputPorts(), destinations, r.outputPorts());
+        }
+        return r;
+    }
+
     void adjustRegions(Vertex sw) {
         List<Region> newRegions = new ArrayList<>();
         for (Region currentRegion : regionsForVertex.get(sw)) {
-            newRegions.addAll(adjustedRegionsFrom(currentRegion));
+            newRegions.addAll(adjustedRegionsFrom(adjustRegionWithOwner(currentRegion, sw)));
```
Comment (rmilfont) [inline src/rbr/RBR.java]: I think that you can rearrange this in order to make more clear (remove the 'if' from adjustRegionWithOwner)

**Review by rmilfont — CHANGES_REQUESTED** [review]
Some little changes

File: src/rbr/RBR.java [hunk 4]
```diff
@@ -129,10 +129,25 @@ public void regionsComputation() {
         assert reachabilityIsOk();
     }
 
+    private boolean isOwnerInside(Region r, Vertex owner) {
```
Comment (jmferreiratech) [inline src/rbr/RBR.java]: Check if the region's outsiders contains the 'owner' instead.

File: tests/rbr/adjustRegionWithOwnerTest.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        ip.add('I');
+        op.add('N');
+        op.add('E');
+        dst.add(new Vertex("0.3"));
+        dst.add(new Vertex("1.3"));
+        dst.add(new Vertex("1.2"));
+    }
+
+    @Test
+    public void adjustedRegionsOwnerInsideRegion() throws Exception {
```
Comment (jmferreiratech) [inline tests/rbr/adjustRegionWithOwnerTest.java]: I'd add a test showing that the adjustedRegion is the same original region when the owner is not inside the box.

File: tests/rbr/adjustRegionWithOwnerTest.java [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+        Region region = new Region(ip, dst, op);
+        Vertex owner = new Vertex("0.2");
+        Region adjustedRegion = rbr.adjustRegionWithOwner(region, owner);
+        Range box = adjustedRegion.box();
+        Assert.assertTrue(owner.isIn(box));
```
Comment (jmferreiratech) [inline tests/rbr/adjustRegionWithOwnerTest.java]: I'd assert only that the adjustedRegion's outsiders is empty.

File: tests/rbr/adjustRegionWithOwnerTest.java (hunk 5 above)
Comment (jmferreiratech) [inline tests/rbr/adjustRegionWithOwnerTest.java]: Your test name should be 'sentence like', stating something that is true. I'd name this one 'adjustedRegionContainsOwner' or something like that.

File: tests/rbr/adjustRegionWithOwnerTest.java [hunk 7]
```diff
@@ -0,0 +1,36 @@
+package rbr;
+
+import java.util.*;
+import org.junit.*;
+import util.*;
+
+public class adjustRegionWithOwnerTest {
+    Set<Character> ip = new HashSet<>();
```
Comment (jmferreiratech) [inline tests/rbr/adjustRegionWithOwnerTest.java:8]: Couldn't these be private?

**Review by jmferreiratech — CHANGES_REQUESTED** [review]

**Comment by rafaelgmota** [comment]
Done

**Review by rmilfont — APPROVED** [review]

File: src/rbr/RBR.java [hunk 8]
```diff
@@ -129,10 +129,20 @@ public void regionsComputation() {
         assert reachabilityIsOk();
     }
 
+    Region adjustRegionWithOwner(Region r, Vertex owner) {
+        if(r.outsiders().contains(owner) && r.outsiders().size() == 1) {
+            Set<Vertex> destinations = r.destinations();
+            destinations.add(owner);
```
Comment (jmferreiratech) [inline src/rbr/RBR.java:135]: Make the 'if' surround only the line 135.

File: tests/rbr/adjustRegionWithOwnerTest.java [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+        dst.add(new Vertex("2.3"));
+        Region region = new Region(ip, dst, op);
+        Vertex owner = new Vertex("0.2");
+        Region adjustedRegion = rbr.adjustRegionWithOwner(region, owner);
+        Assert.assertFalse(adjustedRegion.destinations().contains(owner));
+        Assert.assertTrue(adjustedRegion.outsiders().contains(owner));
+    }
+
+    @Test
+    public void adjustedRegionIsTheSameWhenOwnerAreNotInsideTheBox() throws Exception {
```
Comment (jmferreiratech) [inline tests/rbr/adjustRegionWithOwnerTest.java]: Rename to 'adjustedRegionIsTheSameWhenOwnerIsNotInsideBox'.

File: tests/rbr/adjustRegionWithOwnerTest.java [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+        ip.add('I');
+        op.add('N');
+        op.add('E');
+        dst.add(new Vertex("0.3"));
+        dst.add(new Vertex("1.3"));
+        dst.add(new Vertex("1.2"));
+    }
+
+    @Test
+    public void adjustedRegionContainsOwner() throws Exception {
```
Comment (jmferreiratech) [inline tests/rbr/adjustRegionWithOwnerTest.java]: Rename to 'adjustedRegionIncludesOwnerWhenItIsTheOnlyOutsider'.

File: tests/rbr/adjustRegionWithOwnerTest.java [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+        Region region = new Region(ip, dst, op);
+        Vertex owner = new Vertex("0.2");
+        Region adjustedRegion = rbr.adjustRegionWithOwner(region, owner);
+        Assert.assertTrue(adjustedRegion.destinations().contains(owner));
+        Assert.assertTrue(adjustedRegion.outsiders().size() == 0);
```
Comment (jmferreiratech) [inline tests/rbr/adjustRegionWithOwnerTest.java]: Use assertEquals.

File: tests/rbr/adjustRegionWithOwnerTest.java [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+    }
+
+    @Test
+    public void adjustedRegionNotContainsOwner() throws Exception {
```
Comment (jmferreiratech) [inline tests/rbr/adjustRegionWithOwnerTest.java:35]: Rename to 'adjustedRegionDoNotIncludeOwnerWhenThereAreOthersOutsiders'.

**Review by jmferreiratech — CHANGES_REQUESTED** [review]
Please add an ignored test to the case which even with others outsiders the owner should be included to decrease the number of regions.

**Comment by rafaelgmota** [comment]
Done

File: tests/rbr/adjustRegionWithOwnerTest.java [hunk 13]
```diff
@@ -0,0 +1,66 @@
...
+    public void adjustedRegionIncludesOwnerWhenItIsTheOnlyOutsider() throws Exception {
+        Region region = new Region(ip, dst, op);
+        Vertex owner = new Vertex("0.2");
+        Region adjustedRegion = rbr.adjustRegionWithOwner(region, owner);
+        Assert.assertTrue(adjustedRegion.destinations().contains(owner));
+        Assert.assertEquals(adjustedRegion.outsiders().size(), 0);
+    }
+
+    @Test
+    public void adjustedRegionNotContainsOwner() throws Exception {
```
Comment (jmferreiratech) [inline tests/rbr/adjustRegionWithOwnerTest.java:35]: Make this test name more 'sentence like'... which case are you trying to show?

File: tests/rbr/adjustRegionWithOwnerTest.java (hunk 7 above)
Comment (jmferreiratech) [reply tests/rbr/adjustRegionWithOwnerTest.java:8]: No?

**Review by jmferreiratech — CHANGES_REQUESTED** [review]

**Comment by jmferreiratech** [comment]
Already merged.

**Pull request closed without merging**